handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -226,16 +226,13 @@ SSLEngineResult unwrap(SslHandler handler, ByteBuf in, int readerIndex, int len,
             }
 
             @Override
-            int calculateOutNetBufSize(SslHandler handler, int pendingBytes) {
+            int calculateOutNetBufSize(SslHandler handler, int pendingBytes, int numComponents) {
                 return handler.maxPacketBufferSize;
             }
         };
 
         static SslEngineType forEngine(SSLEngine engine) {
-            if (engine instanceof OpenSslEngine) {
-                return TCNATIVE;
-            }
-            return JDK;
+            return engine instanceof ReferenceCountedOpenSslEngine ? TCNATIVE : JDK;
```

Comment:
> JDK SslEngine when using OpenSslEngine and finalizer?

I don't understand this statement. Can you explain.

Basically `ReferenceCountedOpenSslEngine` is the base class of `OpenSslEngine` and is the class which provides the interface which `SslEngineType#TCNATIVE` depends upon.